site/src/common/blocks/DamVideoBlock.tsx

```diff
@@ -1,18 +1,60 @@
 import { PreviewSkeleton, PropsWithData, withPreview } from "@comet/cms-site";
-import { DamVideoBlockData } from "@src/blocks.generated";
-import * as React from "react";
+import { DamImageBlockData, DamVideoBlockData } from "@src/blocks.generated";
+import { VideoPreviewImage } from "@src/common/helpers/VideoPreviewImage";
+import { useState } from "react";
+import styled, { css } from "styled-components";
+
+// TODO: use aspectRatio and previewImage from the block data when available
+interface DamVideoBlockProps extends PropsWithData<DamVideoBlockData> {
+    previewImage?: DamImageBlockData;
+    aspectRatio?: string;
+    sizes?: string;
+}
 
 export const DamVideoBlock = withPreview(
-    ({ data: { damFile, autoplay, showControls } }: PropsWithData<DamVideoBlockData>) => {
+    ({ data: { damFile, autoplay, loop, showControls }, previewImage, aspectRatio = "auto", sizes = "100vw" }: DamVideoBlockProps) => {
         if (damFile === undefined) {
             return <PreviewSkeleton type="media" hasContent={false} />;
         }
 
+        const hasPreviewImage = previewImage && previewImage.block?.props.damFile;
+        const [showPreviewImage, setShowPreviewImage] = useState(true);
+
         return (
-            <video autoPlay={autoplay} controls={showControls} playsInline muted={autoplay}>
-                <source src={damFile.fileUrl} type={damFile.mimetype} />
-            </video>
+            <>
+                {hasPreviewImage && showPreviewImage && (
+                    <VideoPreviewImage
+                        onClick={() => setShowPreviewImage(false)}
+                        image={previewImage}
+                        aspectRatio={aspectRatio !== "auto" ? aspectRatio : undefined}
+                        sizes={sizes}
+                    />
+                )}
+                {(!showPreviewImage || !hasPreviewImage) && (
```

Comment:
These conditions can be combined
https://github.com/vivid-planet/comet-starter/pull/268#discussion_r1652683960

```suggestion
                {hasPreviewImage && showPreviewImage ? (
                    <VideoPreviewImage
                        onClick={() => setShowPreviewImage(false)}
                        image={previewImage}
                        aspectRatio={aspectRatio !== "auto" ? aspectRatio : undefined}
                        sizes={sizes}
                    />
                ) : (
```

Reply:
yep, thats true 👍